### PR TITLE
Refactored for better readability, added useful extensions and changed activation method on NSLayoutConstraint 

### PR DESCRIPTION
**Pull request by diogoAntunes**
Added:
- IntExtension for syntactic sugar in repeating a closure (used in **ImageStackView**);
- UICollectionViewExtension to make use of generics in order to register and dequeue a cell;
- UICollectionViewCellExtension to conform to a protocol assuring a reusable identifier for the cells;

Refactored:
- struct Dimensions present in most classes changed to an enum to assure better readability and less error prone, since empty enums cannot be created and empty structs can;
- removed “**CameraView**” conformance to the “**CLLocationManagerDelegate**” since this moved to it's own class "LocationManager";
- “**CameraView**” extension to conform to the "**CameraManDelegate**" for better readability and separation of responsibilities;
- renamed lazy vars to a new syntax sugar that assures better readability, so instead of this:

``` swift
public lazy var doneButton: UIButton = { [unowned self] in
    let button = UIButton()
    button.setTitle(Configuration.cancelButtonTitle, forState: .Normal)
    button.titleLabel?.font = Configuration.doneButton
    button.addTarget(self, action: #selector(doneButtonDidPress(_:)), forControlEvents: .TouchUpInside)

    return button
    }()
```

We can change it to:

``` swift
  public lazy var doneButton: UIButton = { [unowned self] in
    $0.setTitle(Configuration.cancelButtonTitle, forState: .Normal)
    $0.titleLabel?.font = Configuration.doneButton
    $0.addTarget(self, action: #selector(doneButtonDidPress(_:)), forControlEvents: .TouchUpInside)

    return $0
    }(UIButton())
```

In my opinion it leaves the namespace more clean and concise and there is no confusion like the first example where there is a button called '_doneButton_' but then the one being returned is called '_button_' :bowtie:

Finally changed the way NSLayoutConstraint was being activated from:

``` swift
addConstraint(NSLayoutConstraint(item: pickerButton, attribute: attribute,
        relatedBy: .Equal, toItem: self, attribute: attribute,
        multiplier: 1, constant: 0))
```

to:

``` swift
NSLayoutConstraint(item: pickerButton, attribute: attribute,
        relatedBy: .Equal, toItem: self, attribute: attribute,
        multiplier: 1, constant: 0).active = true
```

Since the older method is going to be deprecated.

``` swift
public func addConstraint(constraint: NSLayoutConstraint) // This method will be deprecated in a future release and should be avoided.  Instead, set NSLayoutConstraint's active property to YES.
```

### CODE REVIEW

File: Source/ImageGallery/ImageGalleryView.swift [hunk 1]
```diff
@@ -30,51 +30,46 @@ public class ImageGalleryView: UIView {
...
-    }()
+    $0.font = Configuration.noImagesFont
+    $0.textColor = Configuration.noImagesColor
+    $0.text = Configuration.noImagesTitle
+    $0.alpha = 0
+    $0.sizeToFit()
+    self.addSubview($0)
+
+    return $0
+    }(UILabel())
```
Comment (onmyway133) [inline Source/ImageGallery/ImageGalleryView.swift]: 👍

File: Source/BottomView/StackView.swift [hunk 2]
```diff
@@ -7,22 +7,22 @@ protocol ImageStackViewDelegate: class {
 
 class ImageStackView: UIView {
 
-  struct Dimensions {
+  enum Dimensions {
```
Comment (onmyway133) [inline Source/BottomView/StackView.swift]: While empty enum cannot be created, enum itself represents a choice. I'd prefer a struct here

**Comment by onmyway133** [comment]
👍

**Comment by RamonGilabert** [comment]
Any update here? :)

**Comment by diogoAntunes** [comment]
Hey, dunno :) do I need to make some changes for the pull request to be accepted? :)

**Comment by RamonGilabert** [comment]
@onmyway133 had a comment I believe, and there's merge conflicts! After that I believe we can merge :)

**Comment by diogoAntunes** [comment]
Regarding the comment from @onmyway133 I can change the enum to a struct :)
On the merging conflicts I have no idea where can I see them. Should I clone the latest branch and try to apply these pull request and solve possible merge issues? Sorry for the inexperience, it is my first pull request :bowtie:

**Comment by RamonGilabert** [comment]
Oh! Basically you'll pull from master and have conflicts that you'll have to fix. I believe it will be not that many issues. Once you have everything ready, a push to the branch should fix the merge conflicts in GitHub! :)

Once you pull from master, tell me if you have lots of issues marked.

**Comment by diogoAntunes** [comment]
Ok @RamonGilabert should be up and running ;)

**Comment by RamonGilabert** [comment]
Hey @diogoAntunes, before going forward, I've seen that your configuration in Xcode has a 4, 6 or 8 spacing setting, causing a problem with the current setup of 2 spaces that we have for tabs. Could you fix that?

Sorry for the inconvenience.

**Comment by diogoAntunes** [comment]
I think that is now fixed 😄

File: Source/BottomView/BottomContainerView.swift [hunk 3]
```diff
@@ -2,117 +2,112 @@ import UIKit
...
 
-    UIView.animateWithDuration(0.3, animations: {
-      imageView.transform = CGAffineTransformMakeScale(1.05, 1.05)
-      }) { _ in
-        UIView.animateWithDuration(0.2) { _ in
-          imageView.transform = CGAffineTransformIdentity
-        }
-    }
-  }
+		UIView.animateWithDuration(0.3, animations: {
```
Comment (onmyway133) [inline Source/BottomView/BottomContainerView.swift]: It would be nice if we follow the project indentation

**Comment by onmyway133** [comment]
@diogoAntunes Hi, thanks for contributing 💪  Unfortunately, there's some unresolved merge conflicts, and as our code base has been migrated to Swift 3, this can't be merged any more.

**Pull request closed without merging**